samples/snippets/src/main/java/pubsub/SubscribeWithExactlyOnceConsumerWithResponseExample.java

```diff
@@ -0,0 +1,85 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsub;
+
+// [START pubsub_subscriber_exactly_once]
+// [START pubsub_create_subscription_with_exactly_once_delivery]
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class SubscribeWithExactlyOnceConsumerWithResponseExample {
+    public static void main(String... args) throws Exception {
+        // TODO(developer): Replace these variables before running the sample.
+        String projectId = "your-project-id";
+        String subscriptionId = "your-subscription-id";
+
+        subscribeWithExactlyOnceConsumerWithResponseExample(projectId, subscriptionId);
+    }
+
+    public static void subscribeWithExactlyOnceConsumerWithResponseExample(String projectId, String subscriptionId) {
+        ProjectSubscriptionName subscriptionName =
+                ProjectSubscriptionName.of(projectId, subscriptionId);
+
+        // Instantiate an asynchronous message receiver.
+        MessageReceiverWithAckResponse receiverWithResponse =
+                (PubsubMessage message, AckReplyConsumerWithResponse consumerWithResponse) -> {
+                    // Handle incoming message, then ack the received message, and receive the response
+                    System.out.println("Id: " + message.getMessageId());
+                    System.out.println("Data: " + message.getData().toStringUtf8());
+                    Future<AckResponse> ackResponseFuture = consumerWithResponse.ack();
+
+                    // Retreive the completed future
+                    AckResponse ackResponse = ackResponseFuture.get();
+
+                    // Parse the response
+                    switch (ackResponse) {
+                        case AckResponse.SUCCESSFUL:
+                            System.out.println("Message successfully acked");
+                            break;
+                        case AckResponse.INVALID:
+                        case AckResponse.OTHER:
+                            System.out.println("Message failed to acked with response: {0}", ackResponse);
+                            break;
+                        default:
+                            break;
+                    }
+                };
+
+        Subscriber subscriber = null;
+        try {
+            subscriber = Subscriber.newBuilder(subscriptionName, receiverWithResponse)
+                    .setExactlyOnceDeliveryEnabled(true)
+                    .build();
```

Comment:
We should allow setting exactly once delivery on the subscription. Is `.setExactlyOnceDeliveryEnabled()` updating the subscription under the hood to enable exactly once? If so, we need a comment here.

Reply:
Discussed offline. Resolving